samples/apps/copilot-chat-app/webapp/tests/chat.test.ts

```diff
@@ -1,51 +1,239 @@
+    });
+});
+
+async function ServerHealth( page ) {
     // Make sure the server is running.
     await page.goto('https://localhost:40443/healthz');
-    expect(page.getByText('Healthy')).toBeDefined();
-
-    await page.goto('/');
-    // Expect the page to contain a "Login" button.
-    await page.getByRole('button').click();
-    // Clicking the login button should redirect to the login page.
-    await expect(page).toHaveURL(new RegExp('^' + process.env.REACT_APP_AAD_AUTHORITY));
-    // Login with the test user.
-    await page.getByPlaceholder('Email, phone, or Skype').click();
-    await page.getByPlaceholder('Email, phone, or Skype').fill(process.env.REACT_APP_TEST_USER_ACCOUNT as string);
-    await page.getByRole('button', { name: 'Next' }).click();
-    await page.getByPlaceholder('Password').click();
-    await page.getByPlaceholder('Password').fill(process.env.REACT_APP_TEST_USER_PASSWORD as string);
-    await page.getByRole('button', { name: 'Sign in' }).click();
-
-    // Select No if asked to stay signed in.
-    const isAskingStaySignedIn = await page.$$("text='Stay signed in?'");
-    if (isAskingStaySignedIn) {
-        await page.getByRole('button', { name: 'No' }).click();
-    }
-
-    // After login, the page should redirect back to the app.
-    await expect(page).toHaveTitle('Copilot Chat');
-
-    // Send a message to the bot and wait for the response.
-    const responsePromise = page.waitForResponse('**/chat');
-    await page.locator('#chat-input').click();
-    await page.locator('#chat-input').fill('Hi!');
-    await page.locator('#chat-input').press('Enter');
-    await responsePromise;
-
-    // Expect the chat history to contain 3 messages.
-    // The first message is the welcome message from the bot.
-    // The second message is the user's message.
-    // The third message is the bot's response.
+    await expect(page.getByText('Healthy')).toBeDefined();
+}
+
+async function BasicBotResponses( page ) {    
+    await util.LoginAndCreateNewChat(page);
+    
+    const joke = "Can you tell me a funny joke about penguins?";
+    await util.SendChatMessageAndWaitForResponse(page, joke);
+
+    const funfact = "Tell me a fun fact about the cosmos!";
+    await util.SendChatMessageAndWaitForResponse(page, funfact);
+
+    // Expect the chat history to contain 7 messages (both user messages and bot responses).
     const chatHistoryItems = page.getByTestId(new RegExp('chat-history-item-*'));
-    expect((await chatHistoryItems.all()).length).toBe(3);
+    await expect((await chatHistoryItems.all()).length).toBe(5);
+
+    // Expect the last message to be the bot's response.
+    await expect(chatHistoryItems.last()).toHaveAttribute('data-username', 'Copilot');
+
+    await util.PostUnitTest(page);
+}
+
+async function ChatTitleChange(page) {
+    await util.LoginAndCreateNewChat(page);
+    
+    await page.getByRole('button', { name: 'Edit conversation name' }).click();
```

Comment:
using DOM text in our locators can make our tests brittle since the text could change at anytime.
imo we shouldn't be using hardcoded `name`s when calling `getByRole()` and should also stray away from using `getByPlaceholder()`.

using `getByTestId()` is more resilient to UX changes that may be made.

Reply:
Switched over to getByTestId() where possible.